### PR TITLE
Enable details screen for user's notes

### LINKED ISSUE #86: [Enhancement] Improve the list of sent notes
Add to each note preview info about:
- form code
- question code
- date & time not just date
- question text preview 

Tapping a note from the list should show the complete note details in read only mode. Please see figma here https://www.figma.com/file/lww9lcabUpamTZg8zCExI8/MV-2.0-Prototype?node-id=0%3A1

Screenshot from 2020-11-15 23-01-02

Screenshot from 2020-11-15 23-01-09

### PR DESCRIPTION
**Pull request by lukstbit**
### What does it fix?

Closes #86 

Implements all the features from the linked issue. Some notes about the implementation:

- To be able to actually show the files to the user, those need to be available so I removed the code to delete the files when the note upload is successful. If this code is merged, the user will not have any way to delete those files(besides deleting all the app data) so they'll remain on storage
- To avoid complications, my code doesn't show a preview for uploaded videos. Instead I show a notice to the user and the option to see the video in their favorite app:

video_placeholder

I'm keeping it as draft to correct issues that I missed and get feedback.

### How has it been tested?

Manually, tested it for various situations(only images, images and video, only videos etc)

### CODE REVIEW

**Comment by aniri** [comment]
@lukstbit There could be another way to show the sent notes pics and videos without keeping the files which could mean a lot of used storage :/ For the files that are successfully sent to the server, on the reply to the POST action, we receive the urls for the files. Maybe we could save them locally and load those if the file is no longer in local storage? If there is an internet connection, if not, show only the note text + a snackbar that no internet is available or a message instead of the picture. What do you think?

**Comment by lukstbit** [comment]
> What do you think?

Sounds great, I implemented this. I think the implementation is pretty ok and you can review it if you have time.

**Comment by aniri** [comment]
@lukstbit looking good! :rocket: 

Just a few more layout adjustments to be made:
- add some margins to the images in the note details view
- increase left/right margin of the text in the note details view
- the `forms no - question no` text should look the same as the date text

Also, I noticed a weird small glitch: the `forms no - question no` value sometimes disappears, I think it does so after syncing notes with the server.

**Comment by lukstbit** [comment]
I implemented the requested UI changes. 

> Also, I noticed a weird small glitch: the forms no - question no value sometimes disappears, I think it does so after syncing notes with the server.

I wasn't unable to reproduce this. However, while testing, I did notice a bug that prevented the NoteFragment and NoteFragmentDetails to show the correct question code(you can see the bug by going to forms -> selecting a form -> select a question(remember its code) -> swipe left/right to another question -> go to Add Note and add a note -> see that in the notes list the added note shows the code of the initial selected  question and not the code of question the user swiped to). If you still see the bug you mentioned please let me know so I can take another look.

**Comment by aniri** [comment]
hey @lukstbit it seems to crash now if there are no notes stored:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: ro.code4.monitorizarevot, PID: 31493
    java.lang.IllegalStateException: Parcels.unwrap<Question>…able(Constants.QUESTION)) must not be null
        at ro.code4.monitorizarevot.ui.notes.NoteFragment.onViewCreated(NoteFragment.kt:82)
```

**Comment by lukstbit** [comment]
> hey @lukstbit it seems to crash now if there are no notes stored:

I apologize for that, I added tens of notes from the questions list for testing and never once crossed my mind to add it from the option in the forms list. I should be fixed now.

At the moment the form/question codes string is built on the spot so they will not appear in the list show when you add notes from the forms list. If you want them to appear there as well I can store them in the database so they will be available everywhere.

**Comment by aniri** [comment]
> At the moment the form/question codes string is built on the spot so they will not appear in the list show when you add notes from the forms list. If you want them to appear there as well I can store them in the database so they will be available everywhere.

@lukstbit It would be awesome if we could show them everywhere! As it's clearer for the users what they filled in :) But it is not as urgent, so we could make a new issue for this if you do not have time now.

And just one more small nitpick for this issue, could you also make the forms text the same style as the date text here:
Screenshot_20201126-231001_Monitorizare Vot

Thank you!! :rocket:

**Comment by lukstbit** [comment]
@aniri I implemented all of the requested changes, check it out!

File: app/src/main/res/values-ro-rRO/strings.xml [hunk 1]
```diff
@@ -126,8 +126,11 @@
     <string name="select_photo">Selectează fotografia</string>
     <string name="file_manager_missing">Te rugăm să instalezi un manager de fișiere.</string>
     <string name="note_submit">Trimite</string>
-
     <string name="notes_history">Istoricul notelor</string>
+    <string name="note_video_preview_notice">Videoclipul atașat nu are un preview. Apăsați aici pentru a îl vizualiza intr-un player extern!</string>
```
Comment (aniri) [inline app/src/main/res/values-ro-rRO/strings.xml]:
```suggestion
    <string name="note_video_preview_notice">Videoclipul atașat nu are un preview. Apăsați aici pentru a îl vizualiza într-un player extern!</string>
```

File: app/src/main/res/values-ro-rRO/strings.xml [hunk 2]
```diff
@@ -126,8 +126,11 @@
...
+    <string name="note_video_previewer_missing">Nu există aplicații capabile să afișeze acest videoclip!</string>
+    <string name="note_details_missing_internet">Ai nevoie de o conexiune la internet pentru a vedea fișierele atașate ale unei note sincronizate!</string>
+    <string name="note_details_codes">Formular %1$s - Q %2$s</string>
```
Comment (aniri) [inline app/src/main/res/values-ro-rRO/strings.xml]:
```suggestion
    <string name="note_details_codes">Formular %1$s - Întrebare %2$s</string>
```

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -126,8 +126,11 @@
     <string name="select_photo">Select Photo</string>
     <string name="file_manager_missing">Please install a file manager.</string>
     <string name="note_submit">Submit</string>
-
+    <string name="note_video_preview_notice">The attached video doesn\'t have a preview to show. Tap here to see it in an external player!</string>
+    <string name="note_video_previewer_missing">There aren\'t any apps capable of showing the video!</string>
+    <string name="note_details_missing_internet">You need to be connected to the internet to be able to see the attached files of a synchronized note!</string>
     <string name="notes_history">Notes history</string>
+    <string name="note_details_codes">Form %1$s - Q %2$s</string>
```
Comment (aniri) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="note_details_codes">Form %1$s - Question %2$s</string>
```

File: app/src/main/java/ro/code4/monitorizarevot/ui/notes/NoteDetailsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        viewModel.noteDetails.observe(viewLifecycleOwner, Observer {
+            if (it.isSynced) {
+                if (requireActivity().isOnline()) {
+                    noteContentAdapter.updateAdapter(it)
+                } else {
+                    noteContentAdapter.updateAdapter(it.copy(attachedFiles = emptyList()))
+                    Snackbar.make(
+                        view,
+                        getString(R.string.note_details_missing_internet),
+                        Snackbar.LENGTH_SHORT
```
Comment (aniri) [inline app/src/main/java/ro/code4/monitorizarevot/ui/notes/NoteDetailsFragment.kt]:
```suggestion
                        Snackbar.LENGTH_LONG
```

**Review by aniri — APPROVED** [review]

**Pull request merged**